<!doctype html>
<html >
<head>

<script>
	/**
		Block statement
	/**
	JavaScript prior to ECMAScript 6 does not have block scope. Variables introduced within a block are scoped to the containing function or script, and the effects of setting them persist beyond the block itself. In other words, block statements do not define a scope. "Standalone" blocks in JavaScript can produce completely different results from what they would produce in C or Java. For example:
	**/
	var x = 1;
	{
	  var x = 2;
	}
	console.log(x);
	var y;
	console.log(y+2);// "A" - "B" + 
	console.log( "A" - "B" + 2);
	/**
	Switch
	**/
	var fruittype = "Apples";
	switch (fruittype) {
	  case "Oranges":
		console.log("Oranges are $0.59 a pound.");
		break;
	  case "Apples":
		console.log("Apples are $0.32 a pound.");
		break;
	  case "Bananas":
		console.log("Bananas are $0.48 a pound.");
		break;
	  case "Cherries":
		console.log("Cherries are $3.00 a pound.");
		break;
	  case "Mangoes":
		console.log("Mangoes are $0.56 a pound.");
		break;
	  case "Papayas":
		console.log("Mangoes and papayas are $2.79 a pound.");
		break;
	  default:
	   console.log("Sorry, we are out of " + fruittype + ".");
	}
	console.log("Is there anything else you'd like?");
	
	/**
	Exception Handling
	**/
	
	// Create an object type UserException
	function UserException(message) {
	  this.message = message;
	  this.name = "UserException";
	}

	// Make the exception convert to a pretty string when used as a string 
	// (e.g. by the error console)
	UserException.prototype.toString = function() {
	  return this.name + ': "' + this.message + '"';
	}

	// Create an instance of the object type and throw it
	try{
	throw "Error2";
	throw new UserException("Value too high"); //Uncaught UserException: "Value too high"
	}
	catch(e){
		console.log(e);
	}
	//throw {toString: function() { return "I'm an object!"; } };
	
	//
</script>
</head>
<body>

</body>
</html>